Sources/Apollo/NetworkFetchInterceptor.swift

```diff
@@ -29,7 +29,13 @@ public class NetworkFetchInterceptor: ApolloInterceptor, Cancellable {
       return
     }
     
-    self.currentTask = self.client.sendRequest(urlRequest) { result in
+    self.currentTask?.cancel()
```

Comment:
Theoretically this should not be necessary since a new `NetworkFetchInterceptor` is created for each request chain.

Reply:
> since a new NetworkFetchInterceptor is created for each request chain.
This one is interesting, I haven't seen anywhere that this is requirement. When I was migrating at first I just cached all interceptors but then thought that it might be the case that they "have to be new"

Reply:
I added this to the docs on #1484